lib/instrumentation/winston.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const createFormatter = require('@newrelic/winston-enricher/lib/createFormatter')
+const API = require('../../api')
+
+module.exports = function instrument(agent, winston, _, shim) {
+  const config = agent.config
+  if (
+    !(
+      config.application_logging &&
+      config.application_logging.enabled &&
+      config.application_logging.local_decorating &&
+      config.application_logging.local_decorating.enabled
+    )
+  ) {
+    shim.logger.debug('Application logging not enabled. Not instrumenting winston...')
+    return // do not instrument logs if logging is not enabled
```

Comment:
nit: but that debug msg directly above states this

Reply:
👍